gossipd/gossip.c

```diff
@@ -951,11 +945,33 @@ struct returning_peer {
 	struct daemon *daemon;
 	struct pubkey id;
 	struct crypto_state cs;
-	u64 gossip_index;
 	u8 *inner_msg;
 	int peer_fd, gossip_fd;
 };
 
+static void drain_and_forward_gossip(struct peer *peer, int gossip_fd)
```

Comment:
I was briefly afraid that this may introduce an ordering issue: enqueuing a `channel_announcement` while the peer was remote, and the `channel_update` as soon as the peer returned, and then draining into the `msg_queue` would change the order. But I think this is safe, since we only set the flag to reroute gossip messages after draining.